### PR TITLE
ec2_group support for egress rules and referring to security groups by name in rules

### PR DESCRIPTION
**Pull request by esp1**
This patch allows security groups to be referred to by name in rules by using the group_name parameter. If a named security group does not exist, it will be automatically created.

Rules now also support a type parameter to specify whether they are ingress or egress rules. If the type parameter is omitted the rule will be considered an ingress rule.

### CODE REVIEW

File: library/cloud/ec2_group [hunk 1]
```diff
@@ -99,13 +101,120 @@ except ImportError:
     print "failed=True msg='boto required for this module'"
     sys.exit(1)
 
+def lookup_security_group(groups, name, vpc_id):
+    # find the group if present
+    group = None
+
+    for curGroup in groups.values():
+        if curGroup.name == name and curGroup.vpc_id == vpc_id:
+            group = curGroup
```
Comment (pas256) [inline library/cloud/ec2_group:135]: Minor efficiency improvement: break out of the loop once you have found the group.

File: library/cloud/ec2_group [hunk 2]
```diff
@@ -88,9 +88,11 @@ EXAMPLES = '''
         from_port: 10051
         to_port: 10051
         group_id: sg-12345678
-      - proto: all
```
Comment (pas256) [inline library/cloud/ec2_group:113]: Don't delete the only `proto: all` example. The only way to know about this otherwise is to read all the code.

**Comment by pas256** [comment]
Apart from the above, I think this looks good. Main thing to test is to make sure everything works inside and outside of a VPC, referencing by ID and by name.

**Comment by esp1** [comment]
Thanks, I've made the changes you suggested.

**Comment by mpdehaan** [comment]
Hi, sorry for the delay in getting to this one -- we've had quite a backlog.

There are merge commits in this one, which we normally do not accept.

Be sure to always use "git pull --rebase" vs "git pull" and "git rebase" vs "git merge".

We would also prefer to see things as squash commits such that they are easier to review.

Let me know if you would be able to update (or resubmit) this PR.

Thanks!

**Comment by esp1** [comment]
No problem, thanks for getting back to me. I can certainly squash the commits and resubmit. I'll do that tonight. 

Thanks,

Edwin

**Comment by esp1** [comment]
Resubmitted as #6416

**Pull request closed without merging**